src/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -68,46 +98,37 @@ public ConfidentialClientApplicationBuilder WithCertificate(X509Certificate2 cer
         /// <returns></returns>
         public ConfidentialClientApplicationBuilder WithClientSecret(string clientSecret)
         {
+            if (string.IsNullOrWhiteSpace(clientSecret))
+            {
+                throw new ArgumentNullException(nameof(clientSecret));
+            }
+
             Config.ClientSecret = clientSecret;
             return this;
         }
 
-        /// <inheritdoc />
-        internal override void Validate()
+        /// <summary>
+        /// Sets the application client assertion
+        /// </summary>
+        /// <param name="signedClientAssertion">The client assertion used to prove the identity of the application to Azure AD. This is a Base-64 encoded JWT.</param>
+        /// <returns></returns>
+        public ConfidentialClientApplicationBuilder WithClientAssertion(string signedClientAssertion)
         {
-            base.Validate();
```

Comment:
Moving client credential validation logic to the client credential wrapper to consolidate client credential operations.